### ISSUE TITLE
axios_retry_1.default is not a function

### ISSUE THREAD

**Issue by xkizer** [softonic/axios-retry#53]
index.d.ts says that there is a default export, but index.js has no default export, thereby throwing this error when compiled without babel.

**Comment by philjones88** [comment]
The workaround (not pretty) is to use `require` like:

`const axiosRetry = require('axios-retry');`

**Comment by ChavaSobreyra** [comment]
I ended up just doing `axiosRetry.default(...)`

**Comment by threesquared** [comment]
Same issue and also seeing

`Property 'isNetworkOrIdempotentRequestError' does not exist on type 'IAxiosRetry'`

**Comment by mattwexa** [comment]
Hi, I had the same error. I solved it by setting `"esModuleInterop": true` in `tsconfig.json` and using the ES6-style import `import axiosRetry from 'axios-retry';`. This setting is recommended according to the [official TypeScript website](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html)

I'll open an issue for the missing types.

**Comment by mattwexa** [comment]
I think what is missing is something in the README to say that this library requires `esModuleInterop` to be `true`. In that case, I suggest that this issue be closed and I'll create a pull request to add documentation. Does that sound ok? @softonic @philjones88

**Comment by nandafirmans** [comment]
this is works for me

``` typescript
import * as axios from 'axios'

declare function axiosRetry(
  axios: axios.AxiosStatic | axios.AxiosInstance,
  axiosRetryConfig?: axiosRetry.IAxiosRetryConfig
): void

declare namespace axiosRetry {
  interface IAxiosRetryConfig {
    /**
     * The number of times to retry before failing
     * default: 3
     *
     * @type {number}
     */
    retries?: number,
    /**
     * Defines if the timeout should be reset between retries
     * default: false
     *
     * @type {boolean}
     */
    shouldResetTimeout?: boolean,
    /**
     * A callback to further control if a request should be retried. By default, it retries if the result did not have a response.
     * default: error => !error.response
     *
     * @type {Function}
     */
    retryCondition?: (error: axios.AxiosError) => boolean,
    /**
     * A callback to further control the delay between retry requests. By default there is no delay.
     *
     * @type {Function}
     */
    retryDelay?: (retryCount: number, error: axios.AxiosError) => number
  }
}

export = axiosRetry
```

**Comment by dmitrirussu** [comment]
same issue

**Comment by dinvlad** [comment]
Would it be possible to take another look at this? We still get the error `axios_retry_1.default is not a function`. Thanks!

**Comment by mawrkus** [comment]
Hi guys, my TS knowledge is limited... Does one of these PRs fix the issue:

- https://github.com/softonic/axios-retry/pull/81
- https://github.com/softonic/axios-retry/pull/92

?

If not, any PR is welcome!

**Comment by mawrkus** [comment]
https://github.com/softonic/axios-retry/pull/92 has been merged.

**Comment by jugglingthebits** [comment]
I still get the error `axios_retry_1.default is not a function` at runtime.

**Comment by JCass45** [comment]
I am also getting this error when `esModuleInterop: true` is not used. Closing my initial issue #143 in favour of this one.

**Comment by lucasnishimura** [comment]
I still get the error "axios_retry_1.default is not a function" too

**Comment by MaksimKiselev** [comment]
**TL;DR**

Solutions for TypeScript users:
- Set `"esModuleInterop": true` in `tsconfig.json` and use the ES6-style import `import axiosRetry from 'axios-retry';` (refactoring required for all imports)
- Use `require` `const axiosRetry = require('axios-retry');` (no need refactoring)

**Comment by frosas** [comment]
Another ugly hack in case you don't want to lose the type checking:

```js
import importedAxiosRetry from 'axios-retry';

const axiosRetry: typeof importedAxiosRetry = require('axios-retry');
```

**Comment by slinkardbrandon** [comment]
My team ran into this error today as well when trying to use the library. Our "solution" is below:

```diff
- import axiosRetry, { exponentialDelay, isNetworkOrIdempotentRequestError } from 'axios-retry';
+ import { exponentialDelay, isNetworkOrIdempotentRequestError } from 'axios-retry';
+ const axiosRetry = require('axios-retry');
```

I poked around the *axios-retry* source a little bit today too. I ran the `build` command in the `package.json` file, it generates a `lib` directory with an `index.js` file, that `lib/index.js` file then gets required by the project root `index.js`.

The post-babel output `lib/index.js` file declares a `exports.default` along with a few others..

```
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isNetworkError = isNetworkError;
exports.isRetryableError = isRetryableError;
exports.isSafeRequestError = isSafeRequestError;
exports.isIdempotentRequestError = isIdempotentRequestError;
exports.isNetworkOrIdempotentRequestError = isNetworkOrIdempotentRequestError;
exports.exponentialDelay = exponentialDelay;
exports.default = axiosRetry;

// ... more code

function axiosRetry(axios, defaultOptions) {
  axios.interceptors.request.use(function (config) {
    var currentState = getCurrentState(config);
    currentState.lastRequestTime = Date.now();
    return config;
  });

  // ... more code
}

// ... more code
```

I forked the repo thinking this would be a super simple adjustment that I could patch up but the variable and function declarations are really strange to me and I had a hard time wrapping my head around them. 

I would probably propose changing the whole export structure & stop adding object properties to a function under the hood, but that feels extreme for what should have been a really simple fix. Does anyone have any additional thoughts or guidance here?

**Comment by rastilin** [comment]
@slinkardbrandon 

Thanks for your post. Our Cordova project doesn't have a defined .ts file to apply settings but your import method worked to solve the problem.

**Comment by mindhells** [comment]
This issue should be fixed with the latest updates.
Please, feel free to reopen it in case you experience this issue again.
